source/mir/random/discrete.d

```diff
@@ -88,29 +84,31 @@ struct Discrete(T)
     size_t opCall(RNG)(ref RNG gen) const
     {
         import std.random : uniform;
+        import std.range : assumeSorted;
+
         T v = uniform!("[)", T, T)(0, _cdPoints[$-1], gen);
```

Comment:
@WebDrake there are (at least) two ways to do this:

#### inclusive uniform interval & less sortedness

```d
T v = uniform!"[]"(0, cdPoints[$-1], gen);
return cdPoints.assumeSorted!("a < b").lowerBound(v).length;
```
#### semi-inclusive uniform interval & less-or-equal sortedness

```d
T v = uniform!"[)"(0, cdPoints[$-1], gen);
return cdPoints.assumeSorted!("a <= b").lowerBound(v).length;
```